config/config.go

```diff
@@ -524,6 +525,30 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("stored_requests.http_events.amp_endpoint", "")
 	v.SetDefault("stored_requests.http_events.refresh_rate_seconds", 0)
 	v.SetDefault("stored_requests.http_events.timeout_ms", 0)
+	v.SetDefault("stored_video.filesystem.enabled", false)
+	v.SetDefault("stored_video.filesystem.path", "")
+	v.SetDefault("stored_video.directorypath", "./stored_requests/data/by_id")
```

Comment:
Good catch, thanks.

The query would be the same form as the normal fetcher, SELECT id, data, datatype FROM .... where the details would depend on the implementation of the data store. `datatype` would always be `request` unless we want to refactor the fetcher setup so we have `imp`s being pulled from the same query as video requests. I would be against that, as then the stored `imp`s could not also be used on openrtb side. Well, it would probably be possible with a fancy enough query, but the current architecture enforces that the same datastore provides video imps and openrtb imps.

Reply:
The other way to go is to have the main fetcher return three types, "request", "imp", and "video" and do away with a separate video fetcher. But that path leads to havign a fetcher that returns "request", "imp", "video", "amp", "category". I think it's probably better to have smaller targeted, semi-redundant configs than a huge all-in-one fetcher that demands support for use cases not all hosts will have.

Reply:
I was able to make it work with this query: `SELECT aa.uuid, aa.config as data, 'request' as dataType FROM s2sconfig_config aa WHERE aa.uuid IN %REQUEST_ID_LIST%;` (should  it be accounts_account table?).
I like the idea of `FetchRequests()` refactoring, because it becomes more complicated and tangled.

Reply:
Do recall if we make the postgres fetcher one big multipurpose fetcher, we must make all fetchers into big multipurpose fetchers. Otherwise the config will become much much more complicated and tangled.

Also are you saying a big multipurpose fetcher is more complicated and tangled, or simple single purpose fetchers are more complicated and tangled?

Reply:
I'm ok with current implementation. 
It's just becomes more complicated to figure out how does it work, what parameter to pass in case we need stored requests, stored imps, video stored requests and categories.  
For the DB fetcher I would like to see simple single select query rather than complicated multi select and union query and then separate parser for data type ("request" or "imp"). 
I understand it's now a trade of between complicated config and complicated fetcher. 
I think if we will have more data to fetch in future we will need to refactor big fetcher to simple single purpose fetchers.

Reply:
Oh I mean separate fetchers by purpose, like: stored request fetcher, stored impressions fetcher and video stored request fetcher. So DB fetcher for each of those will have a simple sql query and there will no mess with which parameter to pass. 
```FetchRequests(ctx context.Context, requestIDs []string)```
and 
```FetchImpressions(ctx context.Context, requestIDs []string, impIDs []string))```
and
```FetchVideoRequests(ctx context.Context, videoRequestIDs []string)``` (if we need to store in separate table )
instead of
```FetchRequests(ctx context.Context, requestIDs []string, impIDs []string))```

Reply:
Sorry, I expressed myself unclear in previous comment. I meant refactor fetchers functions, not fetchers itself

Reply:
`Perhaps if the list of fetchers continues to expand it will be worth simplifying the fetcher in this manner.` - this is exactly what I meant!